src/contexts/ErrorContext/ErrorContext.tsx

```diff
@@ -1,85 +1,69 @@
 import Prompt from "components/Prompt";
-import { GENERIC_ERROR_MESSAGE } from "constants/common";
-import { APError, AP_ERROR_DISCRIMINATOR } from "errors/errors";
+import { EMAIL_SUPPORT } from "constants/env";
 import { logger } from "helpers";
-import {
-  PropsWithChildren,
-  createContext,
-  useCallback,
-  useContext,
-} from "react";
+import { ReactNode } from "react";
 import { useModalContext } from "../ModalContext";
 
-type State = {
-  handleError: (
-    error: any,
-    displayMessage?: string,
-    options?: { log: boolean }
-  ) => void;
-};
+function parseError(error: unknown): string | undefined {
+  if (typeof error === "string") return error;
 
-const Context = createContext<State>({
-  handleError: (_: any, _1?: string, _2?: { log: boolean }) => {},
-});
+  if (typeof error === "object" && error != null) {
+    if ("message" in error) return parseError(error.message);
+    if ("data" in error) return parseError(error.data);
+    if ("error" in error) return parseError(error.error);
+  }
 
-export default function ErrorContext(props: PropsWithChildren<{}>) {
+  if (error instanceof Error) return error.message;
+}
+
+export function useErrorContext() {
   const { showModal } = useModalContext();
 
   /**
-   * biome-ignore lint/correctness/useExhaustiveDependencies: bug in Biome with recursive dependencies
-   * Created an issue on their GH, see https://github.com/biomejs/biome/issues/2361
+   * @description for expected errors
+   * @param message - user can do something about
+   * @param error - if provided, would be logged to sentry for further investigation
    */
-  const handleError: State["handleError"] = useCallback(
-    (error: any, displayMessage?: string, options = { log: true }) => {
-      if (options.log) {
-        logger.error(error);
-      }
+  function displayError(message: ReactNode, error?: unknown) {
```

Comment:
```suggestion
  function displayError(message: ReactNode) {
```
There's no need for `displayError` to *log* the error.

Reply:
TLDR; would revert, no errors that this function is used for needs to be logged 

I initially thought that unlike `handleError` where the primary goal is to capture error and side effect is to display it, `displayError` on the other hand would display the expected error with the option to log that error so that we devs could look into the error's details.